kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -510,25 +526,35 @@ def kube_job_failed(self, metric, scraper_config):
 
     def kube_job_status_failed(self, metric, scraper_config):
         for sample in metric.samples:
+            job_ts = 0
             tags = [] + scraper_config['custom_tags']
             for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
                 if label_name == 'job' or label_name == 'job_name':
                     trimmed_job = self._trim_job_tag(label_value)
+                    job_ts = self._extract_job_timestamp(label_value)
                     tags.append(self._format_tag(label_name, trimmed_job, scraper_config))
                 else:
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
-            self.job_failed_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
+            if job_ts != 0 and job_ts not in self.failed_job_counts[frozenset(tags)].last_jobs_ts:
+                print("Add value to fail")
+                self.failed_job_counts[frozenset(tags)].count += sample[self.SAMPLE_VALUE]
+                self.failed_job_counts[frozenset(tags)].last_jobs_ts.append(job_ts)
```

Comment:
That's better because this gives an accurate count, but now this `last_jobs_ts` grows unbounded, increasing memory usage over time.

To highlight that you can add a `print(len(self.failed_job_counts[frozenset(tags)].last_jobs_ts))` and run the tests with new job executions. I think the list will grow every time the cronjob triggers and you get a new timestamp.